pkg/controller/migplan/migplan_controller.go

```diff
@@ -434,6 +451,50 @@ func (r *ReconcileMigPlan) AddRunningLabelForPlan(plan *migapi.MigPlan) error {
 	return nil
 }
 
+func (r *ReconcileMigPlan) AddUpdateLabelFailedMigration(plan *migapi.MigPlan) error {
+
+	failedMigrations := 0
+	needsUpdate := false
+	migrations, err := plan.ListMigrations(r)
+	if err != nil {
+		return liberr.Wrap(err)
+	}
+
+	//Check if there are any associated migrations that are failed
+	for _, m := range migrations {
+		if m.Status.HasCondition(migctl.PlanNotReady) {
+			failedMigrations++
+		}
+	}
+
+	switch {
+	//There are no failed migrations associated with the Migplan
```

Comment:
the general convention is to write comments for switch case inside the case

Reply:
Done